tests/serverpod_test_server/test_e2e/websocket_auth_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Given a websocket that requires authentication', () {
+    TestAuthKeyManager authKeyManager = TestAuthKeyManager();
+    late Client client;
+
+    setUp(() async {
+      client = Client(
+        serverUrl,
+        authenticationKeyManager: authKeyManager,
+      );
+    });
+
+    tearDown(() async {
+      client.close();
+    });
+
+    test(
+        'when sending a message to a restricted endpoint without authentication '
+        'then the message should be ignored', () async {
+      await client.openStreamingConnection(
+        disconnectOnLostInternetConnection: false,
+      );
+
+      // This should be ignored by the server as user isn't authenticated.
+      await client.signInRequired.sendStreamMessage(SimpleData(num: 666));
+      var streamingStream = client.signInRequired.stream.timeout(
+        const Duration(seconds: 2),
+      );
+      var receivedData = streamingStream.single;
+
+      expect(receivedData, throwsA(isA<TimeoutException>()));
+    });
+
+    test(
+        'when sending a message to a restricted endpoint with authentication '
+        'then the message should succeed', () async {
+      var response =
+          await client.authentication.authenticate('test@foo.bar', 'password');
```

Comment:
Suggestion: Please add clean up for this as well.

I gave a suggestion for how this can be done in a previous comment.